### PR TITLE
feat(calcite-label): adds no-margin property. also updates label-text spacing

### PR DESCRIPTION
**Pull request by asangma**
**Related Issue:** #916

## Summary
* Adds no-margin property
* Updates label-text spacing.

### CODE REVIEW

File: src/components/calcite-label/calcite-label.tsx [hunk 1]
```diff
@@ -34,6 +34,9 @@ export class CalciteLabel {
   @Prop({ mutable: true, reflect: true }) layout: "inline" | "inline-space-between" | "default" =
     "default";
 
+  /** Turn off margin around the label */
+  @Prop() noMargin?: boolean;
```
Comment (driskull) [inline src/components/calcite-label/calcite-label.tsx:38]: maybe `disable-margin`, `frameless` or something less prop based and more based on functionality/intention?

Comment (paulcpederson) [reply src/components/calcite-label/calcite-label.tsx:38]: If we change this, we should probably also change `noPadding` on `calcite-modal`

Comment (macandcheese) [reply src/components/calcite-label/calcite-label.tsx:38]: Yes let's just make sure Calcite-modal and Calcite-loader "no-padding" end up matching. Disable-margin/padding?

**Comment by eriklharper** [comment]
I'm wondering if instead of having a `default` layout, the label styles its contents in the vertical direction by default, applying a `margin-bottom` on `.calcite-label-text` _only_ when there are other nodes present after it.

This kind of approach would address @AdelheidF 's concern as well as address the issue I'm running into when using `calcite-label` internally with `calcite-radio-button` where I need that margin to be `0` without having to create a new `no-margin` prop.

**Comment by eriklharper** [comment]
Also, for browsers that support it we could use flex `gap` with a `margin-bottom` fallback for IE11 and others.

File: src/components/calcite-label/calcite-label.tsx (hunk 1 above)
Comment (asangma) [reply src/components/calcite-label/calcite-label.tsx:38]: I agree. I never like it to be that specific.
TBH, I was thinking of just tightening up the bottom margin on label...at least making it scale dependent. That might help the case that Adelheid is dealing with.

Comment (macandcheese) [reply src/components/calcite-label/calcite-label.tsx:38]: That's added here: https://github.com/Esri/calcite-components/pull/914/files#diff-ad18c8f8961f619818ad69c1d1c15b17R9 , scale-dependent bottom margin. But I think we'd still need a "disable-margin" option

Comment (driskull) [reply src/components/calcite-label/calcite-label.tsx:38]: It does sound a bit too specific about what it is doing under the hood. Something more general would be nice.

**Comment by macandcheese** [comment]
I think adding `disable-margin` (and then updating any instances of `no-padding` -> `disable-padding`) solves for the time being. Longer term we could think about providing these spacing accommodations in another way, like a `calcite-form` component that would space all slotted `calcite-labels` except the last instance, or in another way.

File: src/components/calcite-label/calcite-label.tsx (hunk 1 above)
Comment (paulcpederson) [reply src/components/calcite-label/calcite-label.tsx:38]: Maybe `compact`? Could be a boolean that means "take extra space away".

Comment (macandcheese) [reply src/components/calcite-label/calcite-label.tsx:38]: Compact kind of seems a bit close to scale=s... we already have no-padding, I agree disable-margin/padding is better than that. 

`disable-spacing` could be used on all three components.

 Although that could insinuate it's disabling internal spacing...

**Comment by asangma** [comment]
How about `disable-spacing` and then let the component decide if that's padding, margin, gap, elbow-room, or whatever?

**Comment by macandcheese** [comment]
Looks good - Should we update the components that use "no-padding" to match this?

**Pull request merged**